CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java

```diff
@@ -75,57 +92,65 @@ public void playerJoinEvent(PlayerJoinEvent event) {
             }
         }
 
-        System.out.println("#### Player " + player.getName() + " entered ");
+        if (!plugin.getPlayerUpdater().playersWithScoreboard.contains(player.getUniqueId())) {
+            Utils.debug("#### Player " + player.getName() + " entered to lobby");
+            plugin.myBukkit.runTask(player, null, null, () -> player.teleportAsync(plugin.getLobbyLocation()));
 
-        if (plugin.getCSPlayers().size() >= MAX_PLAYERS) {
-            PacketUtils.sendTitleAndSubtitle(player, ChatColor.YELLOW + "Welcome to CSMC World", ChatColor.RED + "The game is full, please try again later.", 1, 4, 1);
-            return;
-        }
+            if (plugin.getCSPlayers().size() >= MAX_PLAYERS) {
+                PacketUtils.sendTitleAndSubtitle(player, ChatColor.YELLOW + "Welcome to CSMC World", ChatColor.RED + "The game is full, please try again later.", 1, 4, 1);
+            }
 
-        PacketUtils.sendTitleAndSubtitle(player, ChatColor.YELLOW + "Welcome to CSMC World", ChatColor.RED + "Left click to join game.", 1, 4, 1);
+            PacketUtils.sendTitleAndSubtitle(player, ChatColor.GOLD + "Welcome to CSMC World", ChatColor.RED + "Left click to join game.", 1, 4, 1);
 
-        if (config.contains("lobby-location")) {
-            Location lobbyLoc = plugin.getLobbyLocation();
-            player.setFallDistance(1);
-            player.teleport(lobbyLoc);
-            player.setGameMode(GameMode.SURVIVAL);
-            player.getInventory().clear();
-            player.getInventory().setArmorContents(null);
-            player.setFoodLevel(8); //era 6
-            //player.setExhaustion(20);
-            player.getAttribute(Attribute.GENERIC_MAX_HEALTH).setBaseValue(40);
-            player.setHealth(player.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue());
+            if (plugin.getLobbyLocation() != null) {
+                player.setFallDistance(1);
+                player.setGameMode(GameMode.SURVIVAL);
+                player.getInventory().clear();
+                player.getInventory().setArmorContents(null);
+                player.setFoodLevel(8); //era 6

```

Comment:
```suggestion
                player.setFoodLevel(8); //was 6

```

or even remove this comment

```suggestion
                player.setFoodLevel(8);

```